### PR TITLE
feat: add configurable options for each preprocessor

### PR DESCRIPTION
**Pull request by megion**

### CODE REVIEW

File: lib/html2js.js [hunk 1]
```diff
@@ -20,17 +20,39 @@ var escapeContent = function(content) {
...
+function extend(target) {
+  var sources = [].slice.call(arguments, 1);
+  for (var i=0; i<sources.length; i++) {
+    var src = sources[i];
+    for (var prop in src) {
+      target[prop] = src[prop];
+    }
+  }
+  return target;
+}
```
Comment (Dignifiedquire) [inline lib/html2js.js]: Please use `lodash.extend`, you can pull it directly from https://github.com/karma-runner/karma/blob/master/lib/helper.js

**Comment by Dignifiedquire** [comment]
Thanks for the contribution. Could you explain a little bit what issues this solves, and also add some tests for this?

**Comment by megion** [comment]
Hello Friedel,
I have applied the method lodash.extend.

I need this functionality in order to select in each of my tests, which the preprocessor will use. For example I have two types of templates: standard templates in folder 'simple_templates' and the templates were pre-compiled 'compiled_templates'. In each test I want to be able to select what templates will be used. 

Example configuration
```js
// karma.conf.js
module.exports = function(config) {
  config.set({
    preprocessors: {
      'src/simple_templates/**/*.html': ['htmlProcessor1'],
      'src/compiled_templates/**/*.html': ['htmlProcessor2']
    },

    files: [
      '**/*.html'
    ],

    customPreprocessors: {
      htmlProcessor1: {
        base: 'ng-html2js-configurable',
        stripPrefix: 'src/simple_templates/',
        moduleName: 'template.simple'
      },
      htmlProcessor2: {
        base: 'ng-html2js-configurable',
        stripPrefix: 'src/compiled_templates/',
        moduleName: 'template.compiled'
      }
    }
  });
};
```

**Comment by megion** [comment]
I'll write the tests a little later

**Pull request closed without merging**